src/bitmessagekivy/tests/test_create_random_address.py

```diff
@@ -23,38 +21,42 @@ def test_landing_screen(self):
-        # Click on "Proceed Next" Button to "generate random label for address" screen
-        # Some widgets cannot renders suddenly and become not functional so we used loop with a timeout.
-        self.assertExists("//Screen[0]//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]", timeout=5)
+        self.assertExists(
+            '''//Login//Screen[@name=\"check_screen\"]'''
+            '''//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]''', timeout=5
+        )
         # Clicking on Proceed Next Button to redirect to "random" screen
-        self.cli.wait_click('//Screen[0]//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=5)
+        self.cli.wait_click(
+            '''//Login//Screen[@name=\"check_screen\"]'''
+            '''//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]''', timeout=5
+        )
         self.assertExists("//ScreenManager[@current=\"random\"]", timeout=5)
 
     @ordered
     def test_generate_random_address_label(self):
         """Creating New Adress For New User."""
         # Checking the Button is rendered
         self.assertExists(
-            '//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
+            '//Random//RandomBoxlayout//BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=1
+        )
         # Click on Label Text Field to give address Label
         self.cli.wait_click(
-            '//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
+            '//Random//RandomBoxlayout/BoxLayout[0]/AnchorLayout[1]/MDTextField[@hint_text=\"Label\"]', timeout=2)
         # Enter a Label Randomly
         random_label = ""
         for _ in range(10):
             random_label += choice(ascii_lowercase)
-            self.cli.setattr('//RandomBoxlayout//AnchorLayout[1]/MDTextField[0]', "text", random_label)
+            self.cli.setattr('//Random//RandomBoxlayout//AnchorLayout[1]/MDTextField[0]', "text", random_label)
             self.cli.sleep(0.1)
         # Checking the Button is rendered
-        self.assertExists('//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
+        self.assertExists('//Random//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
         # Click on Proceed Next button to generate random Address
-        self.cli.wait_click('//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
+        self.cli.wait_click('//Random//RandomBoxlayout//MDFillRoundFlatIconButton[@text=\"Proceed Next\"]', timeout=3)
         # Checking "My Address" Screen after creating a address
         self.assertExists("//ScreenManager[@current=\"myaddress\"]", timeout=5)
         # Checking the new address is created
-        self.assertExists('//MDList[0]/CustomTwoLineAvatarIconListItem', timeout=10)
+        self.assertExists('//MyAddress//MDList[0]/CustomTwoLineAvatarIconListItem', timeout=10)
```

Comment:
the `MDList` should have and identifier, not an index

Reply:
Something like:
`//MyAddress//MDList[@id="ml"]/CustomTwoLineAvatarIconListItem[0]`

Reply:
I debugged MDList, yes we can add like `name:mdlist_name` but can not get with `self.cli.getattr()`.
I tried this approach with `id` and `name` but it is not working.
`//MyAddress//MDList[@name="mdlist_name"]/CustomTwoLineAvatarIconListItem[0]`